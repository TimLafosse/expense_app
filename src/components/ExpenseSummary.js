import React from 'react';
import { connect } from 'react-redux';
import { getExpensesTotal } from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import texpenses from '../tests/fixtures/expenses';

export const ExpenseSummary = (props) => (
   <div>Viewing {props.expenses.length} expenses totalling {getExpensesTotal(props.expenses)}</div>
   );

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
 