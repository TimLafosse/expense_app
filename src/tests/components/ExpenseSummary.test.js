import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('count and total 2 expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('count and total 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('count and total 0 expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});