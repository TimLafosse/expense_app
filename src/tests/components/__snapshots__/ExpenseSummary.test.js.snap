// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`count and total  expense 1`] = `
<div>
  Viewing 
  0
   expenses totalling 
  0
</div>
`;

exports[`count and total 1 expense 1`] = `
<div>
  Viewing 
  0
   expenses totalling 
  0
</div>
`;

exports[`count and total 2 expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="Rent"
    id="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    note=""
  />
</div>
`;
